src/components/Trade/hooks/useSwapper/useSwapper.ts

```diff
@@ -112,7 +114,14 @@ export const useSwapper = () => {
     TradeAsset | undefined,
     Trade<KnownChainIds>,
   ]
-  const [swapperManager] = useState<SwapperManager>(getSwapperManager())
+
+  const [swapperManager, setSwapperManager] = useState<SwapperManager>(() => new SwapperManager())
```

Comment:
not sure i agree with this. it should probably now have a type sig `<SwapperManager | null>` and initialize to `null` and consumers will need to check for existence?

this is kind of breaking the singleton pattern if we instantiate it but don't add the swappers?

Reply:
True. if I add | null we will have to add some chcks in various places calling to check for the null value.

IMO its fine as-is but if you feel strongly about this I can change it

Reply:
as discussed this is ok with the comment and makes the consumption nicer, and gets us the async initialize